### PR TITLE
Version bump to `v0.5.0`

### PR DESCRIPTION
**Pull request by bharat-thotakura**

### CODE REVIEW

**Comment by richrines1** [comment]
let's also add `type_checking_exempt_modules = ["collections.abc"]` to pyproject.toml in this pr

**Comment by bharat-thotakura** [comment]
> let's also add `type_checking_exempt_modules = ["collections.abc"]` to pyproject.toml in this pr

Sure thing, thanks

File: checks-superstaq/checks_superstaq/checks-pyproject.toml [hunk 1]
```diff
@@ -23,6 +23,8 @@ extend-ignore = [
   "I201",
   "I202",
 ] # these errors conflict with black and/or isort
+type_checking_exempt_modules = ["collections.abc"]
+
```
Comment (richrines1) [inline checks-superstaq/checks_superstaq/checks-pyproject.toml:27]: nit
```suggestion
```

**Comment by richrines1** [comment]
if possible, let's get #841 in before this deploy

**Review by richrines1 — APPROVED** [review]
> if possible, let's get #841 in before this deploy

nvm let's just get this in. lgtm!

**Pull request merged**